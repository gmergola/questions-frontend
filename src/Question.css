.Question-title {
  -webkit-text-stroke: 1.5px rgb(253, 179, 42);
  margin-top: 50px;
  margin-left: 100px;
  margin-right: 60px;
  border-bottom: 5px solid rgb(253, 179, 42);
  font-size: 50px;
  font-family: Impact, Haettenschweiler, 'Arial Narrow Bold', sans-serif;
  color: black;
}

.Question-new-button {
  margin-left: 60px;
  margin-top: 60px;
  border-radius: 20px;
  width: 200px;
  height: 30px;
  border: none;
}

.Question-new-button:hover {
  opacity: 0.5;
}

@media only screen and (max-width: 900px){
  .Question-title {
    -webkit-text-stroke: 1px white;
    margin-top: 20px;
    margin-left: 50px;
    font-size: 30px;
  }

  .Question-new-button {
    margin-left: 25px;
    margin-top: 30px;
    border-radius: 20px;
    width: 150px;
  }

}