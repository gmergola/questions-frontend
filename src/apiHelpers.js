import axios from "axios";
import { BASE_URL } from './config.js';

class FamilyFeudApi {

  /*getQuestions: gets all Questions*/
  static async getQuestions() {
    try{
      let response = await axios({url: `${BASE_URL}/questions`, method: 'get'});
      return response.data;
    }catch (err){
      let message = err.response.data.message;
      throw message;
    }
  }

  /*getQuestion: gets question by questionMain*/
  static async getQuestion(questionMain) {
    try{
      let response = await axios({url: `${BASE_URL}/questions/${questionMain}`, method: 'get'});
      return response.data;
    }catch (err){
      let message = err.response.data.message;
      throw message;
    }
  }
}

export default FamilyFeudApi;