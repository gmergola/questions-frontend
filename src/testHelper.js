import React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

/**
 * Helper function to modify the render function in React Testing Library
 * by wrapping rendered components inside of a router.
 *
 * Taken from https://testing-library.com/docs/example-react-router
 *
 * @param {Component} component - React component we want to render
 * @param {Object} param1 - object with route and history keys
 */

export function renderWithRouter(component,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) {
  const Wrapper = ({ children }) => (
    <Router history={history}>{children}</Router>
  );
  return {
    ...render(component, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  };
}