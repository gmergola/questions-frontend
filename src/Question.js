import React from 'react';
import { useParams} from "react-router-dom";
import GuessAnswerForm from './GuessAnswerForm';
import FamilyFeudApi from './apiHelpers';
import useApi from './hooks/useApi';
import { useHistory } from "react-router-dom";
import './Question.css';

/**Question: a component with a route of /questions/:question_main
 * renders the GuessAnswerForm
 * Shows information for that specific question using the url parameter 'question_main'
 */
function Question() {
  const { question_main } = useParams();
  const [loading, data] = useApi(FamilyFeudApi.getQuestion, question_main);
  const history = useHistory();

  function selectANewQuestion() {
    history.push('/');
  }

  return (
    loading ? <p>Loading...</p> :
    <div>
      <button className="Question-new-button" onClick={selectANewQuestion}><b>Select A New Question</b></button>
      <div className="Question-title">{data.question}</div>
      <GuessAnswerForm data={data}/>
    </div>
  );
}

export default Question;