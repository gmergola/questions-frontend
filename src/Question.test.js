import React from "react";
import { renderWithRouter } from './testHelper';
import Question from './Question';
import * as apiFns from './apiHelpers';

const MOCKED_QUESTION_DATA = { question: "question", answers: [{ id: 1, answer: 'answer', vote: 456, options: ['option'] }] };

function setMocks() {
  apiFns.default.getQuestion = jest.fn().mockReturnValue(MOCKED_QUESTION_DATA);
}

function resetMocks() {
  apiFns.default.getQuestion.mockRestore();
}

describe("renders with mocked question", () => {
  beforeAll(() => {
    setMocks();
  });

  it("renders without crashing", function () {
    renderWithRouter(<Question />)
  });

  it("matches snapshot", function () {
    const { asFragment } = renderWithRouter(<Question />);
    expect(asFragment()).toMatchSnapshot();
  });

  afterAll(() => {
    resetMocks();
  });
});
