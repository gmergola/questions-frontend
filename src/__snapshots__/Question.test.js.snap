// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with mocked question matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      class="Question-new-button"
    >
      <b>
        Select A New Question
      </b>
    </button>
    <div
      class="Question-title"
    />
    <div
      class="GuessAnswerForm-container"
    >
      <div
        class="spinner"
      />
    </div>
  </div>
</DocumentFragment>
`;
